src/herder/TxSetFrame.cpp

```diff
@@ -1962,16 +1963,16 @@ TxSetPhaseFrame::txsAreValid(Application& app,
     auto diagnostics = DiagnosticEventManager::createDisabled();
     for (auto const& tx : *this)
     {
-        auto txResult = tx->checkValid(app.getAppConnector(), ls, 0,
-                                       lowerBoundCloseTimeOffset,
-                                       upperBoundCloseTimeOffset, diagnostics);
-        if (!txResult->isSuccess())
+        auto res = app.getHerder().checkValidCached(
```

Comment:
I don't think we can use the cached value here, unless we use the ledger sequence as a part of the cache key.